ProjFS.Mac/PrjFSKextTests/HandleOperationTests.mm

```diff
@@ -706,5 +712,176 @@ - (void) testCloseWithoutModifed {
     XCTAssertFalse(MockCalls::DidCallFunction(ProviderMessaging_TrySendRequestAndWaitForResponse));
 }
 
+- (void) testDeleteWithNoVirtualizationRoot {
+    VirtualizationRoots_Cleanup();
+    testFileVnode->attrValues.va_flags = FileFlags_IsEmpty | FileFlags_IsInVirtualizationRoot;
+    HandleVnodeOperation(
+        nullptr,
+        nullptr,
+        KAUTH_VNODE_DELETE,
+        reinterpret_cast<uintptr_t>(context),
+        reinterpret_cast<uintptr_t>(testFileVnode.get()),
+        0,
+        0);
+
+    XCTAssertFalse(MockCalls::DidCallFunction(ProviderMessaging_TrySendRequestAndWaitForResponse));
+}
+
+- (void) testDeleteWhenRequestFails {
+    ProviderMessageMock_SetDefaultRequestResult(false);
+    testFileVnode->attrValues.va_flags = FileFlags_IsEmpty | FileFlags_IsInVirtualizationRoot;
+    HandleVnodeOperation(
+        nullptr,
+        nullptr,
+        KAUTH_VNODE_DELETE,
+        reinterpret_cast<uintptr_t>(context),
+        reinterpret_cast<uintptr_t>(testFileVnode.get()),
+        0,
+        0);
+
+   XCTAssertTrue(MockCalls::CallCount(ProviderMessaging_TrySendRequestAndWaitForResponse) == 1);
+}
+
+- (void) testDeleteDirectoryWithNoVirtualizationRoot {
+    ProviderMessageMock_SetCleanupRootsAfterRequest(true);
```

Comment:
Should we be calling `VirtualizationRoots_Cleanup();` here instead of ` ProviderMessageMock_SetCleanupRootsAfterRequest(true)`?

I noticed that's what we're doing in the `testDeleteWithNoVirtualizationRoot` and `testReadAttributesDirectoryWithNoVirtualizationRoot` tests.

Reply:
Under the covers two calls to TryGetVirtualizationRoot are made in the code path.

To get full coverage we want the first call to success and the second to fail.  This is why we can't just call 
`VirtualizationRoots_Cleanup();`.  That would make the first call fail never testing the second.

I did think about weather we needed to do this check twice but:
1. Since we're multi-threaded I *think* the state of a VirtualizationRoot can change while processing
2. Perhaps refactoring is a good candidate for a separate PR once we have all the tests in place.

Let me know what you think!

Reply:
>Under the covers two calls to TryGetVirtualizationRoot are made in the code path.
>To get full coverage we want the first call to success and the second to fail.

Ahh okay, in that case could you rename the test?  "WithNoVirtualizationRoot" made me think there'd be no root from the start, perhaps something like `testDeleteDirectoryWithDisappearingRoot`?